core/src/main/scala/ai/forestflow/utils/SourceStorageProtocols.scala

```diff
@@ -85,7 +93,44 @@ object SourceStorageProtocols extends StrictLogging {
   }
 
   case object S3 extends EnumVal {
-    def downloadDirectoryImpl(remotePath: String, localDirectory: File, fqrv: FQRV, sslVerify: Boolean): Unit = ???
+
+    private implicit val actorSystem: ActorSystem = ActorSystem("Application-s3-stream")
+    private implicit val executionContext: ExecutionContext = actorSystem.dispatcher
+    private implicit val materializer: ActorMaterializer = ActorMaterializer()
+
+    private val s3PathStylePathPattern = "/(.*)/(.*)".r
+
+    def downloadDirectoryImpl(remotePath: String, localDirectory: File, fqrv: FQRV, sslVerify: Boolean): Unit = {
+
+      val uri = URI.create(remotePath)
+
+      val (bucket, bucketKey) = uri.getPath match {
+        case s3PathStylePathPattern(bucket, bucketKey) => (bucket, bucketKey)
+        case _ => throw new IllegalArgumentException(s"Invalid url $remotePath, bucket and/or key name information couldn't be extracted from the url's path")
+      }
+
+      val localFilePath = Paths.get(localDirectory.getAbsolutePath, bucketKey)
+
+      val s3Settings = S3Ext(actorSystem).settings
+        .withEndpointUrl(uri.getScheme + "://" + uri.getAuthority)
+        .withPathStyleAccess(true) //this is stop gap solution until we get access to virtual hosting of buckets
```

Comment:
This needs to be inferred based on the pattern match done by s#PathStylePathPatern etc..
Cannot be hard coded to true.